### DISCUSSION TITLE
What are the potential risks of adding new arguments to pre-existing blocks in custom extensions?

### DISCUSSION THREAD

**Discussion by LilyMakesThings** [TurboWarp/extensions#289] [Q&A] [answered]
I understand from reading the documentation that you shouldn't, but what problems can arise from when you do?

**Comment by GarboMuffin** [comment, marked as answer]
existing blocks may have a broken input that user's cant type things into

it is easy anyways to create a new block and reimplement that one in terms of the old one

**Reply by LilyMakesThings** [reply]
Ahh yes, that makes sense. Does this apply to Scratch.ArgumentType.IMAGE?

**Reply by GarboMuffin** [reply]
that should be fine
